nemo/collections/nlp/parts/nlp_overrides.py

```diff
@@ -270,21 +269,8 @@ def optimizer_sharded_state_dict(self):
             key: value for key, value in model_sharded_state_dict.items() if not key.endswith('_extra_state')
         }
 
-        if isinstance(optimizer, MegatronDistributedFusedAdam):
+        if not isinstance(optimizer, MainParamsOptimizerWrapper):
             return optimizer.sharded_state_dict(model_sharded_state_dict)
-        elif not isinstance(optimizer, MainParamsOptimizerWrapper):
```

Comment:
sure about these changes? This part I am not familiar but could affect all other gpt-like models